src/push-manifest.ts

```diff
@@ -41,72 +81,106 @@ const extensionToTypeMapping: {[extname: string]: LinkType} = {
   '.woff': 'font',
   '.woff2': 'font'
 };
-export interface PushManifestEntry {
-  [dependencyAbsoluteUrl: string]: {
-    type?: LinkType,
-    // NOTE(fks) 04-05-2017: Only weight=1 is supported by browsers at the
-    // moment. When support is added, we can add automatic weighting and support
-    // multiple numbers.
+
+/**
+ * Get the default resource type for a file based on its extension.
+ */
+function getResourceTypeFromUrl(url: string): ResourceType|undefined {
+  return extensionToTypeMapping[path.extname(url)] || undefined;
+}
+
+/**
+ * Get the resource type for an import, handling special import types and
+ * falling back to getResourceTypeFromUrl() if none exist.
+ */
+function getResourceTypeFromImport(importFeature: Import): ResourceType|
+    undefined {
+  const importTagName = importFeature.astNode.tagName;
```

Comment:
recommend switching off of checking various importFeature.kinds.has() options instead, as the astNode may not generally be an HTML node.

Hm, arguably the kind of imported resource should probably be something that the Import tells you, rather than you try to deduce.